### PR TITLE
JBPM 8915 : Stunner - Properties panel is being closed when switching assets or views

### PR DESCRIPTION
**Pull request by inodeman**
Hi @romartin can you review this PR? Will upload War in next comment

### CODE REVIEW

**Comment by inodeman** [comment]
Jenkins execute full downstream build

**Comment by inodeman** [comment]
Hi @romartin war was uploaded, can you review it?

File: kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-kogito/kie-wb-common-stunner-kogito-client/src/main/java/org/kie/workbench/common/stunner/kogito/client/screens/DiagramEditorPropertiesScreen.java [hunk 1]
```diff
@@ -59,6 +65,9 @@
     private final SessionManager clientSessionManager;
     private final Event<ChangeTitleWidgetEvent> changeTitleNotificationEvent;
     private final DiagramEditorScreenView view;
+    private final Event<ScreenPreMaximizedStateEvent> screenStateEvent;
+    private final Event<ScreenDiagramModelUnSelectedEvent> screenDiagramModelUnSelectedEventEvent;
```
[the remark this answers is not in the document]
Comment (inodeman) [reply kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-kogito/kie-wb-common-stunner-kogito-client/src/main/java/org/kie/workbench/common/stunner/kogito/client/screens/DiagramEditorPropertiesScreen.java]: My bad @manstis will change it

File: kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-kogito/kie-wb-common-stunner-kogito-client/src/main/java/org/kie/workbench/common/stunner/kogito/client/screens/DiagramEditorPropertiesScreen.java [hunk 2]
```diff
@@ -59,6 +65,9 @@
...
+    private final Event<ScreenDiagramPropertiesSwitchingSessionsEvent> screenDiagramPropertiesSwitchingSessionsEventEvent;
```
[the remark this answers is not in the document]
Comment (inodeman) [reply kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-kogito/kie-wb-common-stunner-kogito-client/src/main/java/org/kie/workbench/common/stunner/kogito/client/screens/DiagramEditorPropertiesScreen.java]: My bad @manstis will change it

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-project-client/src/main/java/org/kie/workbench/common/stunner/bpmn/project/client/editor/BPMNDiagramEditor.java [hunk 3]
```diff
@@ -81,6 +88,10 @@
 
     private Consumer<Boolean> saveCallback;
     private boolean isMigrating = false;
+    private boolean useTimer = true;
```
[the remark this answers is not in the document]
Comment (inodeman) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-project-client/src/main/java/org/kie/workbench/common/stunner/bpmn/project/client/editor/BPMNDiagramEditor.java]: Great @manstis thanks for the tip, will change it

**Comment by inodeman** [comment]
Jenkins execute full downstream build

**Comment by inodeman** [comment]
Jenkins execute full downstream build

**Comment by inodeman** [comment]
Hi @LuboTerifaj I added the changes from the other PR about minimize panel, Explore panel should also work now on this PR

**Comment by mbiarnes** [comment]
Jenkins execute full downstream build

**Comment by inodeman** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
Please @inodeman this need **rebase and rebuilds**, otherwise we cannot proceed with the testing. thansk!

**Comment by inodeman** [comment]
Hi @romartin as requested by @LuboTerifaj I am making changes so that Case Definition Diagrams are also supported, the changes I initially made are for BPMN Diagrams only

**Comment by romartin** [comment]
ok thanks @inodeman , in fact it makes sense all editors behave the same way. But in case issues appears on other editors (CM/DMN) I would suggest creating a new ticket and merging this one, at least, which fixes the concrete issue requested. 
Also please do the rebase and re-fire builds... Thanx!

**Comment by inodeman** [comment]
Very well @romartin will do, also I think it is a good idea that this PR be merged into this one, since some of the code is shared, this particular PR includes everything from https://github.com/kiegroup/kie-wb-common/pull/3127, what do you think?

**Comment by inodeman** [comment]
Jenkins execute full downstream build

**Comment by inodeman** [comment]
Hi @romartin rebased and resolved conflicts now

**Comment by inodeman** [comment]
Jenkins execute full downstream build

**Comment by inodeman** [comment]
Hi @LuboTerifaj artifacts generated, can you check this PR?

**Comment by inodeman** [comment]
Jenkins execute full downstream build

**Comment by inodeman** [comment]
Jenkins retest please

**Comment by inodeman** [comment]
Hi @LuboTerifaj right now on the ticket fix works for BPMN files (Explore Panel not included, as it was not mentioned on ticket, will need additional time), for DMN and Case Management will need more time, you can check the issues with BPMN file and I can create jiras for Case and DMN or I can send the fix in a couple of days, let me know what you think?

**Comment by LuboTerifaj** [comment]
Hello @inodeman the issue is not fixed for Properties panel either.

See my previous [comment](https://github.com/kiegroup/kie-wb-common/pull/3129#pullrequestreview-354299842) for more details.

The first mentioned issue in that comment is also the part of steps to reproduce in [JBPM-8915](https://issues.redhat.com/browse/JBPM-8915) jira.

Hence, the main issue is still not fixed.

**Comment by inodeman** [comment]
Hi @LuboTerifaj no problem, I will add Case Management and DMN, should be done early next week

**Comment by romartin** [comment]
@inodeman please do not forget this one. I know we're focused on kogito, but that was expected for the previous release and it's a customer request as well, so we shoud complete it anyway.
Thanks!

**Comment by inodeman** [comment]
no problem @romartin I can start on this tomorrow, just finished the quickstart examples its updates and the error issues creation

**Comment by inodeman** [comment]
Jenkins execute full downstream build

**Comment by inodeman** [comment]
Hi @LuboTerifaj , @romartin can you review this PR? @LuboTerifaj note this change only applies to bpmn models,  you can test the cases when the user 1) maximizes and minimizes with both explore and properties panel, 2) switches views (Documentation, Overview and back to Diagram) for both the properties and explore, 3) Switching Sessions (Properties and Explore Panels), after you can verify how it is supposed to behave, I will go ahead and include DMN and Case Management models, I think this way, I can code it faster after I get your approval on desired behaviour, let me know

**Comment by LuboTerifaj** [comment]
Jenkins execute full downstream build

**Comment by LuboTerifaj** [comment]
Jenkins execute full downstream build

**Comment by inodeman** [comment]
Hi @LuboTerifaj @romartin this PR is ready for review again.

This will apply to BPMN process and Case Definition Only, since as commented by @romartin there is no case in doing it for Case Management featture.

I fixed some issues in changing tabs, now it should behave as expected. Made a change, where you open Overview or Documentation and Open Properties or Explorer and you change to Documentation or Overview, they will remain open.

War can be found here https://drive.google.com/file/d/1i9Q_J8QxTODQ2c1Nj8z4Xt74U15Q3uFG/view?usp=sharing

**Comment by romartin** [comment]
@inodeman @LuboTerifaj see my comment... https://issues.redhat.com/browse/JBPM-8915?focusedCommentId=14324800&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-14324800

**Comment by inodeman** [comment]
Jenkins retest this

**Comment by inodeman** [comment]
jenkins do fdb

**Comment by inodeman** [comment]
Jenkins retest this

**Comment by inodeman** [comment]
jenkins do fdb

**Comment by inodeman** [comment]
Jenkins retest this

**Comment by inodeman** [comment]
jenkins do fdb

**Comment by romartin** [comment]
Closing - out of date and issue closed

**Pull request closed without merging**